ExomeDepth/run_ExomeDepth.py

```diff
@@ -23,30 +23,46 @@ def valid_read(read):
         return False
 
 
-def get_gender(bam):
+def get_sampleid(sample_id):
+    sample_id = re.split("CM|CF|PF|PM|CO", sample_id)[-1]
+    return sample_id
+
+
+def get_gender_clarity(bam):
+    """Get sample gender from Clarity LIMS."""
+    sample_id = get_sampleid(database.functions.get_sample_id(bam))
+    gender_translation = settings.gender_translation
+    lims_client = Lims(settings.clarity_baseuri, settings.clarity_username, settings.clarity_password)
+    samples = lims_client.get_samples(udf={settings.monster_udf: sample_id})
+    gender_list = []
+    for sample in samples:
+        gender_list.append(gender_translation[sample.udf[settings.geslacht_udf].lower()])
+    if len(set(gender_list)) == 1:
+        return list(set(gender_list))[0]
+    else:
+        return "unkown"
+
+
+def get_gender(bam, force=False):
     """Determine chrX ratio based on read count in bam (excl PAR)."""
     with pysam.AlignmentFile(bam, "rb") as workfile:
         xreads = float(sum([valid_read(read) for read in workfile.fetch(region=settings.locus_x)]))
         total = float(workfile.mapped)
     xratio = float("%.4f" % ((xreads / total) * 100))
+    qc_status_gender = ''
     if xratio >= float(settings.ratio_x[1]):
-        return "female"
+        return "female", qc_status_gender
     elif xratio <= float(settings.ratio_x[0]):
-        return "male"
+        return "male", qc_status_gender
     else:
-        bam_base = os.path.basename(bam)
-        if re.search('[C|P]M', bam_base):
-            print((
-                "Sample {0} has a unknown gender based on chrX reads, but resolved as male based on sampleID"
-            ).format(bam_base))
-            return "male"
-        elif re.search('[C|P]F', os.path.basename(bam)):
-            print((
-                "Sample {0} has a unknown gender based on chrX reads, but resolved as female based on sampleID"
-            ).format(bam_base))
-            return "female"
-        else:
-            sys.exit("Sample {0} has a unknown gender and will not be analysed".format(bam_base))
+        # Query Clarity for gender
+        gender = get_gender_clarity(bam)
+        if gender == "unknown" and force is True:
+            gender = settings.force_gender
+            qc_status_gender = "WARNING:GenderForcedTo{}".format(settings.force_gender)
```

Comment:
Just for your information, not required to change. 
Alternative is using f-strings:
`f"WARNING:GenderForcedTo{settings.force_gender}"`